sdk/identity/identity/review/identity-api-review.md

```diff
@@ -0,0 +1,137 @@
+## API Report File for "@azure/identity"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AccessToken } from '@azure/core-http';
+import { GetTokenOptions } from '@azure/core-http';
+import { ServiceClientOptions } from '@azure/core-http';
+import { TokenCredential } from '@azure/core-http';
+
+export { AccessToken }
+
+// @public
+export class AggregateAuthenticationError extends Error {
+    constructor(errors: any[]);
+    errors: any[];
+}
+
+// @public
+export const AggregateAuthenticationErrorName = "AggregateAuthenticationError";
+
+// @public
+export class AuthenticationError extends Error {
+    constructor(statusCode: number, errorBody: object | string | undefined | null);
+    readonly errorResponse: ErrorResponse;
+    readonly statusCode: number;
+}
+
+// @public
+export const AuthenticationErrorName = "AuthenticationError";
+
+// @public
+export class AuthorizationCodeCredential implements TokenCredential {
+    constructor(tenantId: string, clientId: string, clientSecret: string | undefined, authorizationCode: string, redirectUri: string, options?: IdentityClientOptions);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+    }
+
+// @public
+export type BrowserLoginStyle = "redirect" | "popup";
+
+// @public
+export class ChainedTokenCredential implements TokenCredential {
+    constructor(...sources: TokenCredential[]);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+    }
+
+// @public
+export class ClientCertificateCredential implements TokenCredential {
+    constructor(tenantId: string, clientId: string, certificatePath: string, options?: IdentityClientOptions);
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+    }
+
+// @public
+export class ClientSecretCredential implements TokenCredential {
+    constructor(tenantId: string, clientId: string, clientSecret: string, options?: IdentityClientOptions);
```

Comment:
UserPasswordCredential allows you to pass the tenant ID or the name (assuming this since the name is `tenantIdOrName`). Do these other credential types also support it? (the parameter names seem to indicate they don't).

Reply:
I think they all should work with both based on a quick look through the source, since it is just used to compose a URL. With EnvironmentCredential it works with `AZURE_TENANT_ID` set to the tenant domain name or the UUID, since I think the azure service accepts both URLs.

Reply:
no change needed